src/bin/generate_docs/definition.rs

```diff
@@ -49,6 +38,12 @@ impl TypeDefinition {
         let desc1 = self.description.clone();
         let desc2 = other.description.clone();
 
+        let name = if other.name.is_some() {
+            other.name.clone()
+        } else {
+            self.name.clone()
+        };
```

Comment:
```suggestion
        let name = other.name.or(self.name).clone();
```
Wouldn't sth like this work?

Also can you add comment explaining what other.name and self.name represents, when would you see one specified and when the other?

Reply:
Actually, `self.name` is always `None` because `self` always represents an anonymous type (the method was always used on anonymous types).

I removed the whole method. I think it's easier to understand the code when it's placed directly where it's used.